From 09cebfca94169601568fec60f7a6144130f7fd31 Mon Sep 17 00:00:00 2001
From: Marsell Kukuljevic <marsell@joyent.com>
Date: Wed, 10 Oct 2018 22:47:31 +1300
Subject: [PATCH] TRITON-800 "Bhyve Max Vcpus" should be ignored if "Bhyve
 Capable" is not true Reviewed by: Pedro P. Candel <pedro@joyent.com> Reviewed
 by: Trent Mick <trentm@gmail.com> Approved by: Trent Mick <trentm@gmail.com>

---
 lib/cnapi/server.js | 2 +-
 lib/dapi/server.js  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/cnapi/server.js b/lib/cnapi/server.js
index 18e9920..088c2fc 100644
--- a/lib/cnapi/server.js
+++ b/lib/cnapi/server.js
@@ -200,7 +200,7 @@ var s_sysinfo = {
 		'Setup': base.bool.req,
 		'VM Capable': base.bool.req,
 		'Bhyve Capable': base.bool.opt,
-		'Bhyve Max Vcpus': base.intg.pos.opt,
+		'Bhyve Max Vcpus': base.intg.nonneg.opt,
 		'CPU Type': base.str.nonempty.req,
 		'CPU Virtualization': base.str.req,
 		'CPU Physical Cores': base.intg.pos.req,
diff --git a/lib/dapi/server.js b/lib/dapi/server.js
index 27cc5f5..3ee2db1 100644
--- a/lib/dapi/server.js
+++ b/lib/dapi/server.js
@@ -48,7 +48,7 @@ var s_sysinfo = {
 	'type': 'object',
 	'properties': {
 		'Bhyve Capable': base.bool.opt,
-		'Bhyve Max Vcpus': base.intg.pos.opt,
+		'Bhyve Max Vcpus': base.intg.nonneg.opt,
 		'CPU Total Cores': base.intg.pos.req,
 		'HVM API': base.bool.opt,
 		'Network Interfaces': s_si_nics_r,
-- 
2.21.0

